test/test_sparse.py

```diff
@@ -4761,6 +4768,88 @@ def test_gradcheck_mm(self, layout, dtype, device, masked, fast_mode):
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        "empty_sparse_compressed expected sparse compressed [(]non-block[)] tensor layout but got SparseBsr"):
+                    result = op.op(t_inp, *t_args, **t_kwargs)
+            elif op.name == 'mul' and layout is torch.sparse_bsc and t_args[0].ndim > 0:
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        "empty_sparse_compressed expected sparse compressed [(]non-block[)] tensor layout but got SparseBsc"):
+                    result = op.op(t_inp, *t_args, **t_kwargs)
+            elif (op.name == 'mul' and dtype == torch.bool and t_args[0].ndim > 0 and t_inp.is_cpu and t_inp.numel() > 0
+                  and t_inp.dense_dim() > 0):
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        "\"addcmul_cpu_out\" not implemented for 'Bool'"):
+                    result = op.op(t_inp, *t_args, **t_kwargs)
+            elif op.name == 'mul' and dtype == torch.bool and t_args[0].ndim > 0 and t_inp.is_cpu and t_inp.numel() > 0:
+                with self.assertRaisesRegex(
+                        RuntimeError,
+                        "\"mul_out_sparse\" not implemented for 'Bool'"):
+                    result = op.op(t_inp, *t_args, **t_kwargs)
+            else:
+                result = op.op(t_inp, *t_args, **t_kwargs)
+
+                # Check rop(inp, ...).shape == inp.shape
+                self.assertEqual(result.shape, t_inp.shape)
+
+                if layout is torch.sparse_coo and t_inp.numel() == 0 and op.name == 'mul' and t_inp.dense_dim() > 0:
+                    # BUG: gh-97627
+                    with self.assertRaisesRegex(
+                            AssertionError,
+                            "Scalars are not equal!"):
+                        self.assertEqual(result.sparse_dim(), t_inp.sparse_dim())
+                else:
+                    # Check rop(inp, ...).sparse_dim() == inp.sparse_dim()
+                    self.assertEqual(result.sparse_dim(), t_inp.sparse_dim())
+
+                    # Check rop(inp, ...).dense_dim() == inp.dense_dim()
+                    self.assertEqual(result.dense_dim(), t_inp.dense_dim())
+
+                # Check invariant rop(inp, ...).to_dense() == rop(inp.to_dense(), ...)
+                try:
+                    dense = op.op(t_inp.to_dense(), *(t_args[0].to_dense(), *t_args[1:]), **t_kwargs)
+                except Exception as msg:
+                    # this is strided op issue, so skipping the sample silently here
+                    if "\"cpublas_axpy_impl\" not implemented for 'ComplexHalf'" in str(msg):
+                        continue
+                    raise
+                self.assertEqual(result, dense)
+
+        if count == 0:
+            # we count samples to avoid false-positive test reports
+            self.skipTest('no sample inputs')
```

Comment:
Shouldn't we fail here since that seems like a setup issue? Skips often get ignored because most people just look for a boolean signal from CI.

Reply:
Yes, we can fail here.

Reply:
Fixed: replaced `count`-related code with the `non_empty_sampler` wrapper in `OpInfo.sample_inputs_sparse` method that will raise an exception when a sample generator does not produce any samples.